diff --git a/node_modules/@playwright/test/lib/transform.js b/node_modules/@playwright/test/lib/transform.js
index 9910124..948f529 100644
--- a/node_modules/@playwright/test/lib/transform.js
+++ b/node_modules/@playwright/test/lib/transform.js
@@ -149,6 +149,17 @@ function js2ts(resolved) {
     if (!_fs.default.existsSync(resolved) && _fs.default.existsSync(tsResolved)) return tsResolved;
   }
 }
+
+function mebuild(filename, outfile) {
+  require("esbuild").buildSync({
+    entryPoints: [filename],
+    format: "cjs",
+    outfile,
+    sourcemap: "inline",
+  });
+  return {code: _fs.default.readFileSync(outfile).toString()};
+}
+
 function transformHook(code, filename, moduleUrl) {
   // If we are not TypeScript and there is no applicable preprocessor - bail out.
   const isModule = !!moduleUrl;
@@ -166,7 +177,9 @@ function transformHook(code, filename, moduleUrl) {
     const {
       babelTransform
     } = require('./babelBundle');
-    const result = babelTransform(filename, isTypeScript, isModule, hasPreprocessor ? scriptPreprocessor : undefined, [require.resolve('./tsxTransform')]);
+    const result =filename.endsWith('.tsx') ? 
+    mebuild(filename,codePath)
+    : babelTransform(filename, isTypeScript, isModule, hasPreprocessor ? scriptPreprocessor : undefined, [require.resolve('./tsxTransform')]);
     if (result.code) {
       _fs.default.mkdirSync(_path.default.dirname(cachePath), {
         recursive: true
