diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/cjs/index.d.ts b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/cjs/index.d.ts
index f87764a..20716f0 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/cjs/index.d.ts
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/cjs/index.d.ts
@@ -2478,7 +2478,7 @@ declare class Dragging extends StateNode {
     private complete;
 }
 
-declare class DraggingHandle extends StateNode {
+export declare class DraggingHandle extends StateNode {
     static id: string;
     shapeId: TLShapeId;
     initialHandle: TLHandle;
@@ -4066,46 +4066,46 @@ declare class TickManager {
 /** @public */
 export declare const TLArrowShapeDef: TLShapeDef<TLArrowShape, TLArrowUtil>;
 
-declare class TLArrowTool extends StateNode {
+export declare class TLArrowTool extends StateNode {
     static id: string;
     static initial: string;
-    static children: () => (typeof Idle | typeof Pointing)[];
+    static children: () => [typeof Idle, typeof Pointing];
     shapeType: TLShapeType;
     styles: ("color" | "fill" | "dash" | "size" | "opacity" | "font" | "align" | "geo" | "arrowheadStart" | "arrowheadEnd" | "spline" | "icon" | "labelColor")[];
 }
 
 /** @public */
-export declare class TLArrowUtil extends TLShapeUtil<TLArrowShape> {
+export declare class TLArrowUtil<Shape extends TLArrowShape> extends TLShapeUtil<Shape> {
     static type: string;
     canEdit: () => boolean;
     canBind: () => boolean;
     isClosed: () => boolean;
-    hideResizeHandles: TLShapeUtilFlag<TLArrowShape>;
-    hideRotateHandle: TLShapeUtilFlag<TLArrowShape>;
-    hideSelectionBoundsFg: TLShapeUtilFlag<TLArrowShape>;
-    hideSelectionBoundsBg: TLShapeUtilFlag<TLArrowShape>;
-    defaultProps(): TLArrowShape['props'];
-    getCenter(shape: TLArrowShape): Vec2d;
-    getBounds(shape: TLArrowShape): Box2d;
-    getOutlineWithoutLabel(shape: TLArrowShape): VecLike[];
-    getOutline(shape: TLArrowShape): Vec2dModel[];
-    snapPoints(_shape: TLArrowShape): Vec2d[];
+    hideResizeHandles: TLShapeUtilFlag<Shape>;
+    hideRotateHandle: TLShapeUtilFlag<Shape>;
+    hideSelectionBoundsFg: TLShapeUtilFlag<Shape>;
+    hideSelectionBoundsBg: TLShapeUtilFlag<Shape>;
+    defaultProps(): Shape['props'];
+    getCenter(shape: Shape): Vec2d;
+    getBounds(shape: Shape): Box2d;
+    getOutlineWithoutLabel(shape: Shape): VecLike[];
+    getOutline(shape: Shape): Vec2dModel[];
+    snapPoints(_shape: Shape): Vec2d[];
     private get infoCache();
-    getArrowInfo(shape: TLArrowShape): ArrowInfo | undefined;
-    getHandles(shape: TLArrowShape): TLHandle[];
-    onHandleChange: OnHandleChangeHandler<TLArrowShape>;
-    onTranslateStart: OnTranslateStartHandler<TLArrowShape>;
-    onResize: OnResizeHandler<TLArrowShape>;
-    onDoubleClickHandle: (shape: TLArrowShape, handle: TLHandle) => TLShapePartial<TLArrowShape> | void;
-    hitTestPoint(shape: TLArrowShape, point: VecLike): boolean;
-    hitTestLineSegment(shape: TLArrowShape, A: VecLike, B: VecLike): boolean;
-    render(shape: TLArrowShape): JSX.Element | null;
-    indicator(shape: TLArrowShape): JSX.Element | null;
-    get labelBoundsCache(): ComputedCache<Box2d | null, TLArrowShape>;
-    getLabelBounds(shape: TLArrowShape): Box2d | null;
-    getEditingBounds: (shape: TLArrowShape) => Box2d;
-    onEditEnd: OnEditEndHandler<TLArrowShape>;
-    toSvg(shape: TLArrowShape, font: string, colors: TLExportColors): SVGGElement;
+    getArrowInfo(shape: Shape): ArrowInfo | undefined;
+    getHandles(shape: Shape): TLHandle[];
+    onHandleChange: OnHandleChangeHandler<Shape>;
+    onTranslateStart: OnTranslateStartHandler<Shape>;
+    onResize: OnResizeHandler<Shape>;
+    onDoubleClickHandle: (shape: Shape, handle: TLHandle) => TLShapePartial<Shape> | void;
+    hitTestPoint(shape: Shape, point: VecLike): boolean;
+    hitTestLineSegment(shape: Shape, A: VecLike, B: VecLike): boolean;
+    render(shape: Shape): JSX.Element | null;
+    indicator(shape: Shape): JSX.Element | null;
+    get labelBoundsCache(): ComputedCache<Box2d | null, Shape>;
+    getLabelBounds(shape: Shape): Box2d | null;
+    getEditingBounds: (shape: Shape) => Box2d;
+    onEditEnd: OnEditEndHandler<Shape>;
+    toSvg(shape: Shape, font: string, colors: TLExportColors): SVGGElement;
 }
 
 /** @public */
@@ -4806,7 +4806,7 @@ export declare type TLResizeMode = 'scale_shape' | 'resize_bounds';
 /** @public */
 export declare type TLSelectionHandle = SelectionCorner | SelectionEdge | RotateCorner;
 
-declare class TLSelectTool extends StateNode {
+export declare class TLSelectTool extends StateNode {
     static id: string;
     static initial: string;
     static children: () => (typeof Resizing | typeof Translating | typeof Brushing | typeof DraggingHandle | typeof EditingShape | typeof Idle_9 | typeof PointingCanvas | typeof PointingHandle | typeof PointingResizeHandle | typeof PointingRotateHandle | typeof PointingSelection | typeof PointingShape | typeof Rotating | typeof ScribbleBrushing)[];
@@ -5291,7 +5291,7 @@ export declare type TLWheelEventInfo = TLBaseEventInfo & {
     delta: Vec2dModel;
 };
 
-declare class Translating extends StateNode {
+export declare class Translating extends StateNode {
     static id: string;
     info: TLBaseEventInfo & {
         type: "pointer";
diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/app/App.mjs b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/app/App.mjs
index 41f58f9..51a4f78 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/app/App.mjs
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/app/App.mjs
@@ -2827,7 +2827,28 @@ const _App = class extends EventEmitter {
         newShape.index = index;
         index = getIndexAbove(index);
       }
-      if (TLArrowShapeDef.is(newShape)) {
+
+      // for copy and pasting custom arrows to a new canvas
+      // after migrating to newest tldraw, use isShapeOfType
+      // https://github.com/tldraw/tldraw/blob/ed37bcf54126201f4e1915f25845a623ad6c5174/packages/editor/src/lib/editor/Editor.ts#L7633-L7701
+      // or new API for bindings if available
+      // https://discord.com/channels/859816885297741824/926464446694580275/1190596833349742622
+      const initialStartBindingId =
+        shape?.props?.start?.type === "binding"
+          ? shape.props.start.boundShapeId
+          : undefined;
+      const initialEndBindingId =
+        shape?.props?.end?.type === "binding"
+          ? shape.props.end.boundShapeId
+          : undefined;
+
+      const startBindingShape = shapes.find((s) => s.id === initialStartBindingId);
+      const endBindingShape = shapes.find((s) => s.id === initialEndBindingId);
+
+      const hasCopiedBindingShape =
+        startBindingShape !== undefined || endBindingShape !== undefined;
+
+      if (TLArrowShapeDef.is(newShape) || hasCopiedBindingShape) {
         if (newShape.props.start.type === "binding") {
           const mappedId = idMap.get(newShape.props.start.boundShapeId);
           newShape.props.start = mappedId ? { ...newShape.props.start, boundShapeId: mappedId } : (
diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/app/statechart/TLSelectTool/TLSelectTool.mjs b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/app/statechart/TLSelectTool/TLSelectTool.mjs
index 185d39d..f19078a 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/app/statechart/TLSelectTool/TLSelectTool.mjs
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/app/statechart/TLSelectTool/TLSelectTool.mjs
@@ -40,6 +40,7 @@ class TLSelectTool extends StateNode {
   };
 }
 export {
-  TLSelectTool
+  TLSelectTool,
+  DraggingHandle,
 };
 //# sourceMappingURL=TLSelectTool.mjs.map
diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/hooks/useCanvasEvents.mjs b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/hooks/useCanvasEvents.mjs
index 043e7c0..f480bc6 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/hooks/useCanvasEvents.mjs
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/hooks/useCanvasEvents.mjs
@@ -78,7 +78,7 @@ function useCanvasEvents() {
         ;
         e.isKilled = true;
         document.body.click();
-        preventDefault(e);
+        // preventDefault(e);
       }
       function onTouchEnd(e) {
         ;
diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/index.mjs b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/index.mjs
index 57cb723..4be6ebd 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/index.mjs
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/index.mjs
@@ -3,6 +3,8 @@ export * from "./app/App.mjs";
 export * from "./app/shapeutils/index.mjs";
 export * from "./app/statechart/StateNode.mjs";
 export * from "./app/statechart/TLBoxTool/index.mjs";
+export * from "./app/statechart/TLArrowTool/index.mjs";
+export * from "./app/statechart/TLSelectTool/index.mjs";
 export * from "./app/types/clipboard-types.mjs";
 export * from "./app/types/event-types.mjs";
 export * from "./app/types/history-types.mjs";
diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/utils/assets.mjs b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/utils/assets.mjs
index bc63c45..221a390 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/utils/assets.mjs
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/editor/dist/esm/lib/utils/assets.mjs
@@ -54,41 +54,70 @@ async function getResizedImageDataUrl(dataURLForImage, width, height) {
     img.src = dataURLForImage;
   });
 }
+// async function getMediaAssetFromFile(file) {
+//   return await new Promise((resolve, reject) => {
+//     const reader = new FileReader();
+//     reader.onerror = () => reject(reader.error);
+//     reader.onload = async () => {
+//       let dataUrl = reader.result;
+//       const isImageType = isImage(file.type);
+//       const sizeFn = isImageType ? getImageSizeFromSrc : getVideoSizeFromSrc;
+//       if (file.type === "video/quicktime" && dataUrl.includes("video/quicktime")) {
+//         dataUrl = dataUrl.replace("video/quicktime", "video/mp4");
+//       }
+//       const originalSize = await sizeFn(dataUrl);
+//       const size = containBoxSize(originalSize, { w: MAX_ASSET_WIDTH, h: MAX_ASSET_HEIGHT });
+//       if (size !== originalSize && (file.type === "image/jpeg" || file.type === "image/png")) {
+//         dataUrl = await getResizedImageDataUrl(dataUrl, size.w, size.h);
+//       }
+//       const assetId = TLAsset.createCustomId(getHashForString(dataUrl));
+//       const metadata = await getFileMetaData(file);
+//       const asset = {
+//         id: assetId,
+//         type: isImageType ? "image" : "video",
+//         typeName: "asset",
+//         props: {
+//           name: file.name,
+//           src: dataUrl,
+//           w: size.w,
+//           h: size.h,
+//           mimeType: file.type,
+//           isAnimated: metadata.isAnimated
+//         }
+//       };
+//       resolve(asset);
+//     };
+//     reader.readAsDataURL(file);
+//   });
+// }
 async function getMediaAssetFromFile(file) {
-  return await new Promise((resolve, reject) => {
-    const reader = new FileReader();
-    reader.onerror = () => reject(reader.error);
-    reader.onload = async () => {
-      let dataUrl = reader.result;
-      const isImageType = isImage(file.type);
-      const sizeFn = isImageType ? getImageSizeFromSrc : getVideoSizeFromSrc;
-      if (file.type === "video/quicktime" && dataUrl.includes("video/quicktime")) {
-        dataUrl = dataUrl.replace("video/quicktime", "video/mp4");
-      }
-      const originalSize = await sizeFn(dataUrl);
-      const size = containBoxSize(originalSize, { w: MAX_ASSET_WIDTH, h: MAX_ASSET_HEIGHT });
-      if (size !== originalSize && (file.type === "image/jpeg" || file.type === "image/png")) {
-        dataUrl = await getResizedImageDataUrl(dataUrl, size.w, size.h);
-      }
-      const assetId = TLAsset.createCustomId(getHashForString(dataUrl));
-      const metadata = await getFileMetaData(file);
-      const asset = {
-        id: assetId,
-        type: isImageType ? "image" : "video",
-        typeName: "asset",
-        props: {
-          name: file.name,
-          src: dataUrl,
-          w: size.w,
-          h: size.h,
-          mimeType: file.type,
-          isAnimated: metadata.isAnimated
-        }
-      };
-      resolve(asset);
-    };
-    reader.readAsDataURL(file);
+  const url = await window.roamAlphaAPI.file.upload({
+    file, 
+    toast: {
+      hide: true
+    }
   });
+  const dataUrl = url.replace(/^!\[\]\(/, "").replace(/\)$/, "");
+  const isImageType = isImage(file.type);
+  const sizeFn = isImageType ? getImageSizeFromSrc : getVideoSizeFromSrc;
+  const originalSize = await sizeFn(dataUrl);
+  const size = containBoxSize(originalSize, { w: MAX_ASSET_WIDTH, h: MAX_ASSET_HEIGHT });
+  const assetId = TLAsset.createCustomId(getHashForString(dataUrl));
+  const metadata = await getFileMetaData(file);
+  const asset = {
+    id: assetId,
+    type: isImageType ? "image" : "video",
+    typeName: "asset",
+    props: {
+      name: file.name,
+      src: dataUrl,
+      w: size.w,
+      h: size.h,
+      mimeType: file.type,
+      isAnimated: metadata.isAnimated
+    }
+  };
+  return asset;
 }
 async function getFileMetaData(file) {
   if (file.type === "image/gif") {
@@ -287,12 +316,21 @@ async function createBookmarkShapeAtPoint(app, url, point) {
     }
   });
 }
-function createAssetShapeAtPoint(app, svgString, point) {
+async function createAssetShapeAtPoint(app, svgString, point) {
   const svg = new DOMParser().parseFromString(svgString, "image/svg+xml").querySelector("svg");
   if (!svg) {
     throw new Error("No <svg/> element present");
   }
-  const dataUrl = getSvgAsDataUrlSync(svg);
+
+  const svgBlob = new Blob([svgString], { type: "image/svg+xml" });
+  const file = new File([svgBlob], "image.svg", { type: "image/svg+xml" });
+  
+  const url = await window.roamAlphaAPI.file.upload({
+    file,
+  });
+  // const dataUrl = getSvgAsDataUrlSync(svg);
+  const dataUrl = url.replace(/^!\[\]\(/, "").replace(/\)$/, "");
+
   let width = parseFloat(svg.getAttribute("width") || "0");
   let height = parseFloat(svg.getAttribute("height") || "0");
   if (!(width && height)) {
