diff --git a/node_modules/@tldraw/primitives/dist/esm/lib/freehand/setStrokePointRadii.mjs b/node_modules/@tldraw/primitives/dist/esm/lib/freehand/setStrokePointRadii.mjs
index 0bbfddb..ea8a571 100644
--- a/node_modules/@tldraw/primitives/dist/esm/lib/freehand/setStrokePointRadii.mjs
+++ b/node_modules/@tldraw/primitives/dist/esm/lib/freehand/setStrokePointRadii.mjs
@@ -12,7 +12,8 @@ function setStrokePointRadii(strokePoints, options) {
   } = options;
   const { easing: taperStartEase = EASINGS.easeOutQuad } = start;
   const { easing: taperEndEase = EASINGS.easeOutCubic } = end;
-  const totalLength = strokePoints[strokePoints.length - 1].runningLength;
+  const totalLength = strokePoints[strokePoints.length - 1]?.runningLength || 0;
+  if (!totalLength) return strokePoints;
   let firstRadius;
   let prevPressure = strokePoints[0].pressure;
   let strokePoint;
