diff --git a/node_modules/@tldraw/tlschema/dist/esm/shapes/TLEmbedShape.mjs b/node_modules/@tldraw/tlschema/dist/esm/shapes/TLEmbedShape.mjs
index 0c0783a..1136d46 100644
--- a/node_modules/@tldraw/tlschema/dist/esm/shapes/TLEmbedShape.mjs
+++ b/node_modules/@tldraw/tlschema/dist/esm/shapes/TLEmbedShape.mjs
@@ -78,7 +78,7 @@ const EMBED_DEFINITIONS = [
   {
     type: "tldraw",
     title: "tldraw",
-    hostnames: ["beta.tldraw.com", "lite.tldraw.com"],
+    hostnames: ["tldraw.com", "beta.tldraw.com", "lite.tldraw.com"],
     minWidth: 300,
     minHeight: 300,
     width: 720,
