diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/cjs/index.d.ts b/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/cjs/index.d.ts
index 0f93cb5..0e42591 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/cjs/index.d.ts
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/cjs/index.d.ts
@@ -737,6 +737,7 @@ export declare interface ToolItem {
     meta?: {
         [key: string]: any;
     };
+    style?: CSSProperties
 }
 
 /** @public */
diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/esm/lib/components/Toolbar/Toolbar.mjs b/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/esm/lib/components/Toolbar/Toolbar.mjs
index 8c9610d..03a9b7c 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/esm/lib/components/Toolbar/Toolbar.mjs
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/esm/lib/components/Toolbar/Toolbar.mjs
@@ -151,7 +151,7 @@ const OverflowToolsContent = track(function OverflowToolsContent2({
   toolbarItems
 }) {
   const msg = useTranslation();
-  return /* @__PURE__ */ jsx("div", { className: "tlui-button-grid__four tlui-button-grid__reverse", children: toolbarItems.map(({ toolItem: { id, meta, kbd, label, onSelect, icon } }) => {
+  return /* @__PURE__ */ jsx("div", { className: "tlui-button-grid__four tlui-button-grid__reverse", children: toolbarItems.map(({ toolItem: { id, meta, kbd, label, onSelect, icon, style } }) => {
     return /* @__PURE__ */ jsx(
       M.Item,
       {
@@ -162,7 +162,8 @@ const OverflowToolsContent = track(function OverflowToolsContent2({
         "aria-label": label,
         onClick: onSelect,
         title: label ? `${msg(label)} ${kbd ? kbdStr(kbd) : ""}` : "",
-        icon
+        icon,
+        style
       },
       id
     );
@@ -188,7 +189,8 @@ function ToolbarButton({
       onTouchStart: (e) => {
         e.preventDefault();
         item.onSelect();
-      }
+      },
+      style: item.style,
     },
     item.id
   );
diff --git a/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/esm/lib/hooks/useClipboardEvents.mjs b/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/esm/lib/hooks/useClipboardEvents.mjs
index d6b373d..af74af9 100644
--- a/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/esm/lib/hooks/useClipboardEvents.mjs
+++ b/node_modules/@tldraw/tldraw/node_modules/@tldraw/ui/dist/esm/lib/hooks/useClipboardEvents.mjs
@@ -216,6 +216,14 @@ const handleHtmlString = async (app, html, point) => {
   }
 };
 const handleTextString = async (app, text, point) => {
+
+  const MAX_TEXT_SIZE = 500 * 1024; // 500 KB in bytes
+  if (text.length > MAX_TEXT_SIZE) {
+      console.warn('Text is too large to paste:', text.length, 'bytes');
+      pasteText(app, "Text is too large to paste", point);
+      return;
+    }
+
   const s = text.trim();
   const tldrawContent = text.match(/<tldraw[^>]*>(.*)<\/tldraw>/)?.[1];
   if (tldrawContent) {
